app/scripts/controllers/permissions/selectors.js

```diff
@@ -44,6 +44,57 @@ export const getPermittedAccountsByOrigin = createSelector(
   },
 );
 
+// This will be moved to the @metamask/chain-agnostic-permission package
+/**
+ * Get the permitted accounts for a given scope.
+ *
+ * @param {Pick<Caip25CaveatValue, 'requiredScopes' | 'optionalScopes'>} caip25CaveatValue - The CAIP-25 CaveatValue to get the permitted accounts for
+ * @param {string[]} scopes - The scopes to get the permitted accounts for
+ * @returns {string[]} An array of permitted accounts
+ */
+export const getPermittedAccountsForScopes = (caip25CaveatValue, scopes) => {
+  const scopeAccounts = [];
+
+  scopes.forEach((scope) => {
+    const requiredScope = caip25CaveatValue.requiredScopes[scope];
+    const optionalScope = caip25CaveatValue.optionalScopes[scope];
+    if (requiredScope) {
+      scopeAccounts.push(...requiredScope.accounts);
+    }
+
+    if (optionalScope) {
+      scopeAccounts.push(...optionalScope.accounts);
+    }
+  });
+  return scopeAccounts;
+};
```

Comment:
Moved [here](https://github.com/MetaMask/core/pull/5491/files#diff-3e05548aa13d4e9cc180b8fea8084b3c627294ba7fecd1cdf84156b7b269de5fR138), will replace with reference to this package when available.